pytket/qir/conversion/baseprofileqirgenerator.py

```diff
@@ -0,0 +1,126 @@
+from pytket.circuit import (
+    BitRegister,
+    Command,
+    Conditional,
+)
+
+from .module import tketqirModule
+from .qirgenerator import (
+    AbstractQirGenerator,
+)
+
+
+class BaseProfileQirGenerator(AbstractQirGenerator):
+    """Generate QIR from a pytket circuit."""
+
+    def __init__(
+        self,
+        circuit: Circuit,
+        module: tketqirModule,
+        wasm_int_type: int,
+        qir_int_type: int,
+        wfh: Optional[wasm.WasmFileHandler] = None,
+    ) -> None:
+
+        super().__init__(circuit, module, wasm_int_type, qir_int_type, wfh)
+
+        self.measure_results: list[list] = []
+
+        self.reg_const_list: dict[str, list] = {}
+
+        for creg in self.circuit.c_registers:
+            reg_name = creg[0].reg_name
+            self.reg_const_list[reg_name] = []
+
+        for creg in self.circuit.c_registers:
+            reg_name = creg[0].reg_name
+            for i in range(creg.size):
+                self.reg_const_list[reg_name].append(
+                    self.module.module.add_byte_string(str.encode(f"{reg_name}[{i}]"))
+                )
+
+        entry = self.module.module.entry_block
+        self.module.module.builder.insert_at_end(entry)
+
+        # void __quantum__rt__result_record_output(result)
+        self.record_output_res = self.module.module.add_external_function(
+            "__quantum__rt__result_record_output",
+            pyqir.FunctionType(
+                pyqir.Type.void(self.module.module.context),
+                [
+                    pyqir.result_type(self.module.module.context),
+                    pyqir.PointerType(pyqir.IntType(self.module.module.context, 8)),
+                ],
+            ),
+        )
+
+    def _get_bit_from_creg(self, creg: str, index: int) -> Value:
+        raise ValueError("Classical REG Ops Not available in base profile")
+
+    def _set_bit_in_creg(self, creg: str, index: int, ssa_bit: Value) -> None:
+        raise ValueError("Classical REG Ops Not available in base profile")
+
+    def get_ssa_vars(self, reg_name: str) -> Value:
+        raise ValueError("Classical REG Ops Not available in base profile")
+
+    def _get_i64_ssa_reg(self, reg_name: str) -> Value:
+        raise ValueError("Classical REG Ops Not available in base profile")
+
+    def get_ssa_list(self, reg_name: str) -> list:
+        raise ValueError("Classical REG Ops Not available in base profile")
+
+    def set_ssa_vars(self, reg_name: str, ssa_i64: Value, trunc: bool) -> None:
+        raise ValueError("Classical REG Ops Not available in base profile")
+
+    def _reg2ssa_var(self, bit_reg: BitRegister) -> Value:
+        raise ValueError("Classical REG Ops Not available in base profile")
+
+    def conv_conditional(self, command: Command, op: Conditional) -> None:
+        raise ValueError("Conditional NOT available in base profile")
+
+    def conv_measure(self, bits: list[Bit], qubits: list[Qubit]) -> None:
+
+        qubit_index = qubits[0].index[0]
```

Comment:
Should probably (a) check that `bits` and `qubits` are the same length and (b) handle the case where that length is 0.

Reply:
They should always have exactly one entry, I have added an assert. See [40b2085](https://github.com/CQCL/pytket-qir/pull/177/commits/40b2085eabba5552288d5969a799063d171e15eb)